pkg/config/signer_test.go

```diff
@@ -0,0 +1,289 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package config
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"io"
+	"log"
+	"math/big"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	. "github.com/onsi/gomega"
+)
+
+var publicKey, privateKey []byte
+
+func TestMain(m *testing.M) {
+	publicKey, privateKey = generatePublicAndPrivateKey()
+
+	os.Exit(m.Run())
+}
+
+func TestNewSigner(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		gt := NewGomegaWithT(t)
+		signer, err := NewSigner([]byte(publicKey), []byte(privateKey), "test-msp")
+		gt.Expect(err).NotTo(HaveOccurred())
+		gt.Expect(signer.MSPId()).To(Equal("test-msp"))
+		gt.Expect(signer.Cert().Subject.CommonName).To(Equal("Wile E. Coyote"))
+	})
+
+	tests := []struct {
+		spec        string
+		publicKey   []byte
+		privateKey  []byte
+		mspID       string
+		expectedErr string
+		matchErr    bool
+	}{
+		{
+			spec:        "nil public key",
+			publicKey:   nil,
+			privateKey:  []byte(privateKey),
+			mspID:       "test-msp",
+			expectedErr: "failed to get cert from pem: failed to decode pem bytes: []",
+			matchErr:    true,
+		},
+		{
+			spec:        "invalid public key",
+			publicKey:   []byte("apple"),
+			privateKey:  []byte(privateKey),
+			mspID:       "test-msp",
+			expectedErr: "failed to get cert from pem: failed to decode pem bytes",
+			matchErr:    false,
+		},
+		{
+			spec:        "public key is not a certificate",
+			publicKey:   []byte(privateKey),
+			privateKey:  []byte(privateKey),
+			mspID:       "test-msp",
+			expectedErr: "failed to get cert from pem: failed to parse x509 cert",
+			matchErr:    false,
+		},
+		{
+			spec:        "nil private key",
+			publicKey:   []byte(publicKey),
+			privateKey:  nil,
+			mspID:       "test-msp",
+			expectedErr: "failed to decode private key from pem",
+			matchErr:    true,
+		},
+		{
+			spec:        "empty mspID",
+			publicKey:   []byte(publicKey),
+			privateKey:  []byte(privateKey),
+			expectedErr: "failed to create new signer, mspID can not be empty",
+			matchErr:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.spec, func(t *testing.T) {
+			gt := NewGomegaWithT(t)
+
+			_, err := NewSigner(tc.publicKey, tc.privateKey, tc.mspID)
+			if tc.matchErr {
+				gt.Expect(err).To(MatchError(tc.expectedErr))
+			} else {
+				gt.Expect(err.Error()).To(ContainSubstring(tc.expectedErr))
+			}
+		})
+	}
+}
+
+func TestECDSAPublicKeyImport(t *testing.T) {
+	t.Run("certificate does not contain valid ecdsa publicKey", func(t *testing.T) {
```

Comment:
See comment above as well as for the ones below. If you want to annotate the test you can add the context to the matcher https://onsi.github.io/gomega/#annotating-assertions

Reply:
done